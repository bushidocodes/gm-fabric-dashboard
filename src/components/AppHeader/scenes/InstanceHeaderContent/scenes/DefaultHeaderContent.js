import React from "react";
import { PropTypes } from "prop-types";

import Tab from "components/AppHeader/components/Tab";
import TabNav from "components/AppHeader/components/TabNav";

DefaultHeaderContent.propTypes = {
  headerTabs: PropTypes.array
};

/**
 * Default Header Content
 * @export
 * @param {Object} props - See propTypes
 * @returns JSX.Element
 */
export default function DefaultHeaderContent({ headerTabs }) {
  return (
    <TabNav>
      {headerTabs}
      <Tab href={`/explorer`} icon="Explorer" tabIndex={1} title="Explorer" />
    </TabNav>
  );
}
