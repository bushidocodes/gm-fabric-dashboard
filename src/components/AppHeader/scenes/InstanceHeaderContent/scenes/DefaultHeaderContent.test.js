import React from "react";
import { render } from "enzyme";
import { MemoryRouter, Route } from "react-router";

import state from "json/mockReduxState";
import DefaultHeaderContent from "./DefaultHeaderContent";

// Router is necessary because of the <Tab />'s which need a router context
const RouterWrap = (
  <MemoryRouter>
    <Route
      render={props => (
        <DefaultHeaderContent {...props} metrics={state.instance.metrics} />
      )}
    />
  </MemoryRouter>
);

describe("DefaultHeaderContent", () => {
  test("matches snapshot", () => {
    const tree = render(RouterWrap);
    expect(tree).toMatchSnapshot();
  });
});
