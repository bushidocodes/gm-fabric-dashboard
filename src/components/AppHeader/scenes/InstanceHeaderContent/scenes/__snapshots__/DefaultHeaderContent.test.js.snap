// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHeaderContent matches snapshot 1`] = `
.c2 {
  color: rgb(255,255,255);
  white-space: nowrap;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 4px 0 0;
}

.c2 h1 {
  font-size: 18px;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  line-height: 1;
  margin: 0;
  font-weight: 600;
  text-transform: capitalize;
}

.c5 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c4 {
  z-index: 1;
  height: 24px;
  width: 24px;
  color: currentColor;
  word-spacing: 0;
  vertical-align: middle;
}

.c4:not(:root) {
  overflow: visible;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 25px;
}

.c1 {
  background-color: rgb(45,45,45);
  font-weight: 600;
  -webkit-flex: 1 1 1%;
  -ms-flex: 1 1 1%;
  flex: 1 1 1%;
  margin: 1px 1px 0 0;
  overflow: hidden;
  padding: 4px 8px 8px;
  position: relative;
  -webkit-transition: all 0.15s ease;
  transition: all 0.15s ease;
}

.c1::after {
  border-radius: 1px;
  box-shadow: 0 0 1em 1px rgb(10,171,42);
  color: inherit;
  content: "";
  height: 1px;
  left: 0;
  opacity: 0;
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0;
  z-index: 100;
}

.c1:hover {
  background-color: rgb(54,54,54);
}

.c1:hover:after {
  opacity: 0.5;
}

.c1.active {
  background-color: rgb(77,77,77);
}

.c1.active:after {
  background-color: rgb(84,196,106);
  opacity: 1;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c1:active:after {
  background-color: rgb(84,196,106);
  opacity: 1;
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.c1:focus {
  outline: rgb(10,171,42) auto 3px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  color: rgb(150,150,150);
  background-color: rgba(0,0,0,0.30000000000000004);
  padding: 0 1px 1px 1px;
}

<div
  class="c0"
>
  <a
    aria-current="false"
    class="c1"
    href="/explorer"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        <svg
          aria-labelledby="ariaLabelledby"
          class="c4"
          focusable="false"
          version="1.1"
          viewBox="0 0 24 24"
          xlink="http://www.w3.org/1999/xlink"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            class="c5"
            fill="currentColor"
            title=""
          >
            <title />
            <g
              class="glyph"
              fill="currentColor"
              transform="translate(0 0) scale(1)"
            >
              <path
                d="M8.33638483,18.307209 L6.08293365,20.5606602 L5.02227348,19.5 L7.27572466,17.2465488 C7.59309719,17.6343799 7.94855377,17.9898365 8.33638483,18.307209 Z M6.03114366,15.0598605 L5.88504845,15.3004879 L4,12.9718987 L5.16586731,12.0281013 L5.58305689,12.5434532 C5.5880682,13.4267812 5.74578359,14.2739368 6.03114366,15.0598605 Z M20.5625715,13.0568881 L20.5987088,13.1759286 L20.941843,12.6107664 L22.2240243,13.3892336 L20.138587,16.8240714 L19.8246809,15.7900275 C20.2332591,14.9543591 20.4910375,14.0314977 20.5625715,13.0568881 Z M13.0829337,19 C9.49308278,19 6.58293365,16.0898509 6.58293365,12.5 C6.58293365,8.91014913 9.49308278,6 13.0829337,6 C16.6727845,6 19.5829337,8.91014913 19.5829337,12.5 C19.5829337,16.0898509 16.6727845,19 13.0829337,19 Z M13.0829337,17.5 C15.8443574,17.5 18.0829337,15.2614237 18.0829337,12.5 C18.0829337,9.73857625 15.8443574,7.5 13.0829337,7.5 C10.3215099,7.5 8.08293365,9.73857625 8.08293365,12.5 C8.08293365,15.2614237 10.3215099,17.5 13.0829337,17.5 Z M10.075255,14.9118853 L9.60957463,14.0850721 L9.53044999,12.5423736 L10.9691705,9.47727349 L12.7069714,12.0839749 L11.3865761,12.221457 L13.2013677,7.68447803 L14.5149693,12.2820834 L15.6543782,11.4844971 L16.6379617,12.6995121 L16.472534,14.8791465 L15.368632,13.5155029 L13.650898,14.7179166 L12.9644996,12.315522 L12.2497724,14.1023399 L10.4588959,11.4160251 L11.7618614,11.3186804 L10.075255,14.9118853 Z"
                id="Explorer"
              />
            </g>
          </g>
        </svg>
      </span>
      <h1>
        Explorer
      </h1>
    </div>
  </a>
</div>
`;
